--- BUILD.bazel	2023-05-26 17:34:36.407802028 +0200
+++ BUILD.bazel	2023-05-26 17:35:10.456190917 +0200
@@ -130,23 +130,6 @@
         ],
         "//conditions:default": ["-pthread"],
     }),
-    deps = select({
-        ":has_absl": [
-            "@com_google_absl//absl/debugging:failure_signal_handler",
-            "@com_google_absl//absl/debugging:stacktrace",
-            "@com_google_absl//absl/debugging:symbolize",
-            "@com_google_absl//absl/flags:flag",
-            "@com_google_absl//absl/flags:parse",
-            "@com_google_absl//absl/flags:reflection",
-            "@com_google_absl//absl/flags:usage",
-            "@com_google_absl//absl/strings",
-            "@com_google_absl//absl/types:any",
-            "@com_google_absl//absl/types:optional",
-            "@com_google_absl//absl/types:variant",
-            "@com_googlesource_code_re2//:re2",
-        ],
-        "//conditions:default": [],
-    }),
 )

 cc_library(

