diff --git a/launch/launch/logging/__init__.py b/launch/launch/logging/__init__.py
index 207a8ed..6a07d72 100644
--- a/launch/launch/logging/__init__.py
+++ b/launch/launch/logging/__init__.py
@@ -82,24 +82,6 @@ def _make_unique_log_dir(*, base_path):
             return log_dir
 
 
-def _renew_latest_log_dir(*, log_dir):
-    """
-    Renew the symbolic link to the latest logging directory.
-
-    :param log_dir: the current logging directory
-    :return True if the link was successfully created/updated, False otherwise
-    """
-    base_dir = os.path.dirname(log_dir)
-    latest_dir = os.path.join(base_dir, 'latest')
-
-    if os.path.lexists(latest_dir):
-        if not os.path.islink(latest_dir):
-            return False
-        os.unlink(latest_dir)
-    os.symlink(log_dir, latest_dir, target_is_directory=True)
-    return True
-
-
 class LaunchConfig:
     """Launch Logging Configuration class."""
 
@@ -137,18 +119,6 @@ class LaunchConfig:
             self._log_dir = _make_unique_log_dir(
                 base_path=_get_logging_directory()
             )
-            try:
-                success = _renew_latest_log_dir(log_dir=self._log_dir)
-                if not success:
-                    import warnings
-                    warnings.warn(
-                        'Cannot create a symlink to latest log directory')
-
-            except OSError as e:
-                import warnings
-                warnings.warn(
-                    ('Cannot create a symlink to latest log directory: {}\n')
-                    .format(e))
 
         return self._log_dir
 
