diff --git a/rclcpp/src/rclcpp/typesupport_helpers.cpp b/rclcpp/src/rclcpp/typesupport_helpers.cpp
index 7286c35b..4292ebcc 100644
--- a/rclcpp/src/rclcpp/typesupport_helpers.cpp
+++ b/rclcpp/src/rclcpp/typesupport_helpers.cpp
@@ -23,8 +23,6 @@
 #include <tuple>
 #include <utility>

-#include "ament_index_cpp/get_package_prefix.hpp"
-#include "ament_index_cpp/get_resources.hpp"
 #include "rcpputils/shared_library.hpp"
 #include "rcpputils/find_library.hpp"
 #include "rosidl_runtime_cpp/message_type_support_decl.hpp"
@@ -49,11 +47,6 @@ std::string get_typesupport_library_path(
 #endif

   std::string package_prefix;
-  try {
-    package_prefix = ament_index_cpp::get_package_prefix(package_name);
-  } catch (ament_index_cpp::PackageNotFoundError & e) {
-    throw std::runtime_error(e.what());
-  }

   const std::string library_path = rcpputils::path_for_library(
     package_prefix + dynamic_library_folder,
