diff --git a/rclcpp/include/rclcpp/generic_subscription.hpp b/rclcpp/include/rclcpp/generic_subscription.hpp
index 673712ee..099b5ef1 100644
--- a/rclcpp/include/rclcpp/generic_subscription.hpp
+++ b/rclcpp/include/rclcpp/generic_subscription.hpp
@@ -82,9 +82,9 @@ public:
       *rclcpp::get_typesupport_handle(topic_type, "rosidl_typesupport_cpp", *ts_lib),
       topic_name,
       options.template to_rcl_subscription_options<rclcpp::SerializedMessage>(qos),
-      true),
-    callback_(callback),
-    ts_lib_(ts_lib)
+      true,
+      ts_lib),
+    callback_(callback)
   {
     // This is unfortunately duplicated with the code in subscription.hpp.
     // TODO(nnmm): Deduplicate by moving this into SubscriptionBase.
@@ -159,8 +159,6 @@ private:
   RCLCPP_DISABLE_COPY(GenericSubscription)
 
   std::function<void(std::shared_ptr<rclcpp::SerializedMessage>)> callback_;
-  // The type support library should stay loaded, so it is stored in the GenericSubscription
-  std::shared_ptr<rcpputils::SharedLibrary> ts_lib_;
 };
 
 }  // namespace rclcpp
diff --git a/rclcpp/include/rclcpp/subscription_base.hpp b/rclcpp/include/rclcpp/subscription_base.hpp
index f21f27e8..8060e43f 100644
--- a/rclcpp/include/rclcpp/subscription_base.hpp
+++ b/rclcpp/include/rclcpp/subscription_base.hpp
@@ -29,6 +29,8 @@
 #include "rmw/impl/cpp/demangle.hpp"
 #include "rmw/rmw.h"
 
+#include "rcpputils/shared_library.hpp"
+
 #include "rclcpp/any_subscription_callback.hpp"
 #include "rclcpp/detail/cpp_callback_trampoline.hpp"
 #include "rclcpp/experimental/intra_process_manager.hpp"
@@ -84,7 +86,8 @@ public:
     const rosidl_message_type_support_t & type_support_handle,
     const std::string & topic_name,
     const rcl_subscription_options_t & subscription_options,
-    bool is_serialized = false);
+    bool is_serialized = false,
+    std::shared_ptr<rcpputils::SharedLibrary> ts_lib = {});
 
   /// Destructor.
   RCLCPP_PUBLIC
@@ -526,6 +529,10 @@ public:
   rclcpp::ContentFilterOptions
   get_content_filter() const;
 
+private:
+  // The type support library should stay loaded, so it is stored in the GenericSubscription
+  std::shared_ptr<rcpputils::SharedLibrary> ts_lib_;
+
 protected:
   template<typename EventCallbackT>
   void
diff --git a/rclcpp/src/rclcpp/subscription_base.cpp b/rclcpp/src/rclcpp/subscription_base.cpp
index 300f465a..51f03c50 100644
--- a/rclcpp/src/rclcpp/subscription_base.cpp
+++ b/rclcpp/src/rclcpp/subscription_base.cpp
@@ -39,8 +39,10 @@ SubscriptionBase::SubscriptionBase(
   const rosidl_message_type_support_t & type_support_handle,
   const std::string & topic_name,
   const rcl_subscription_options_t & subscription_options,
-  bool is_serialized)
-: node_base_(node_base),
+  bool is_serialized,
+  std::shared_ptr<rcpputils::SharedLibrary> ts_lib)
+: ts_lib_(ts_lib),
+  node_base_(node_base),
   node_handle_(node_base_->get_shared_rcl_node_handle()),
   node_logger_(rclcpp::get_node_logger(node_handle_.get())),
   use_intra_process_(false),
