diff --git a/rclrs/src/lib.rs b/rclrs/src/lib.rs
index 4924b36..3a22c6d 100644
--- a/rclrs/src/lib.rs
+++ b/rclrs/src/lib.rs
@@ -11,6 +11,7 @@ mod clock;
 mod context;
 mod error;
 mod executor;
+mod logging;
 mod node;
 mod parameter;
 mod publisher;
@@ -38,6 +39,7 @@ pub use clock::*;
 pub use context::*;
 pub use error::*;
 pub use executor::*;
+pub use logging::*;
 pub use node::*;
 pub use parameter::*;
 pub use publisher::*;
diff --git a/rclrs/src/node.rs b/rclrs/src/node.rs
index 97684d6..defd1f3 100644
--- a/rclrs/src/node.rs
+++ b/rclrs/src/node.rs
@@ -440,6 +440,17 @@ impl Node {
     pub fn builder(context: &Context, node_name: &str) -> NodeBuilder {
         NodeBuilder::new(context, node_name)
     }
+
+    /// Returns the logger name of the node.
+    pub fn logger_name(&self) -> &str {
+        let rcl_node = self.handle.rcl_node.lock().unwrap();
+        let name_raw_ptr = unsafe { rcl_node_get_logger_name(&*rcl_node) };
+        if name_raw_ptr.is_null() {
+            return "";
+        }
+        let name_cstr = unsafe { CStr::from_ptr(name_raw_ptr) };
+        name_cstr.to_str().unwrap_or("")
+    }
 }
 
 // Helper used to implement call_string_getter(), but also used to get the FQN in the Node::new()
