diff --git a/rosidl_adapter/rosidl_adapter/action/__init__.py b/rosidl_adapter/rosidl_adapter/action/__init__.py
index 6e8719b..951e3f0 100644
--- a/rosidl_adapter/rosidl_adapter/action/__init__.py
+++ b/rosidl_adapter/rosidl_adapter/action/__init__.py
@@ -17,7 +17,6 @@ from rosidl_adapter.resource import expand_template
 
 
 def convert_action_to_idl(package_dir, package_name, input_file, output_dir):
-    assert package_dir.is_absolute()
     assert not input_file.is_absolute()
     assert input_file.suffix == '.action'
 
diff --git a/rosidl_adapter/rosidl_adapter/msg/__init__.py b/rosidl_adapter/rosidl_adapter/msg/__init__.py
index b02b7b5..b99234c 100644
--- a/rosidl_adapter/rosidl_adapter/msg/__init__.py
+++ b/rosidl_adapter/rosidl_adapter/msg/__init__.py
@@ -17,7 +17,6 @@ from rosidl_adapter.resource import expand_template
 
 
 def convert_msg_to_idl(package_dir, package_name, input_file, output_dir):
-    assert package_dir.is_absolute()
     assert not input_file.is_absolute()
     assert input_file.suffix == '.msg'
 
diff --git a/rosidl_adapter/rosidl_adapter/srv/__init__.py b/rosidl_adapter/rosidl_adapter/srv/__init__.py
index c57b701..a196571 100644
--- a/rosidl_adapter/rosidl_adapter/srv/__init__.py
+++ b/rosidl_adapter/rosidl_adapter/srv/__init__.py
@@ -17,7 +17,6 @@ from rosidl_adapter.resource import expand_template
 
 
 def convert_srv_to_idl(package_dir, package_name, input_file, output_dir):
-    assert package_dir.is_absolute()
     assert not input_file.is_absolute()
     assert input_file.suffix == '.srv'
 
