diff --git a/rust/private/rustc.bzl b/rust/private/rustc.bzl
index 98806f1a..10f07e65 100644
--- a/rust/private/rustc.bzl
+++ b/rust/private/rustc.bzl
@@ -240,6 +240,7 @@ def collect_deps(
                 crate_deps.append(struct(
                     crate_info = dep_variant_info.crate_info,
                     dep_info = dep_variant_info.dep_info,
+                    cc_info = dep_variant_info.cc_info,
                 ))
 
     aliases = {k.label: v for k, v in aliases.items()}
